stravalib/util/limiter.py

```diff
@@ -277,7 +308,7 @@ def _get_wait_time(
         long_usage: int,
         seconds_until_short_limit: int,
         seconds_until_long_limit: int,
-    ) -> float:
+    ) -> float | None:
```

Comment:
@enadeau i adjusted the return here to add an OR in case it was none. Pyright was again flagging this where mypy wasn't. i'm curious about the difference between these two type checkers and why mypy was ok with this whereas pyright was not. looking at the code it feels unlikely to return None  but perhaps ?

Reply:
I've not worked a lot with pyright but it seems like here mypy is outsmarting it because `self.priority` is typed as `Literal["low", "medium", "high"]` mypy know that  you will always return in one of the three blocks of the if statement. Pyrigth do not realize that as it only consider self.priority as being an `str`. Since by default a function return None it says that the return type should be `float | None`

I think we definitely don't want to change the return type to `str | None`. As you noticed, it actually causes problem later down the road when we try to use the result in a function that expect a float.

If you want to silence pyright on this one you can raise a relevant exception after the if statement

Reply:
ahhh you are right. i just was listening to the type checker and not closely evaluating the return possibilities. i'm switching to mypy only in my gui. thank you!